src/widget/wcoverart.cpp

```diff
@@ -129,7 +129,7 @@ void WCoverArt::slotCoverFound(const QObject* pRequestor, int requestReference,
     }
 
     if (pRequestor == this && m_loadedTrack &&
-            m_loadedTrack->getId() == requestReference) {
+            m_loadedTrack->getId().toValue() == requestReference) {
```

Comment:
Due to the typedef in DbId we compare here DbId::value_type with int. 
What about replacing toValue with toInt()?